core/transientstore/store.go

```diff
@@ -98,6 +114,173 @@ func (provider *storeProvider) Close() {
 	provider.dbProvider.Close()
 }
 
+// DeleteStore deletes the TransientStoreProvider for a given ledger.
+func (provider *storeProvider) DeleteStore(ledgerID string) error {
+	return provider.dbProvider.Drop(ledgerID)
+}
+
+func (provider *storeProvider) markStorageForDelete(ledgerID string) error {
+	marked, err := provider.getStorageMarkedForDeletion()
+	if err != nil {
+		return errors.WithMessage(err, "while listing delete marked storage")
+	}
+
+	// don't update if the storage is already marked for deletion.
+	for i := 0; i < len(marked); i++ {
+		if strings.EqualFold(ledgerID, marked[i]) {
+			logger.Infof("transient storage %s was already marked for delete", ledgerID)
+			return nil
+		}
+	}
+
+	marked = append(marked, ledgerID)
+
+	err = provider.markStorageListForDelete(marked)
+	if err != nil {
+		return errors.WithMessagef(err, "while updating storage list %v for deletion", marked)
+	}
+
+	return nil
+}
+
+// Write the UNDER_DELETE ledger set as a JSON array in the transient system catalog.
+func (provider *storeProvider) markStorageListForDelete(storageList []string) error {
+	json, err := json.Marshal(storageList)
+	if err != nil {
+		return errors.WithMessagef(err, "marshalling delete list")
+	}
+
+	db := provider.dbProvider.GetDBHandle(systemNamespace)
+	defer db.Close()
+
+	if err = db.Put(underDeletionKey, json, true); err != nil {
+		return errors.WithMessagef(err, "writing delete list to system storage")
+	}
+
+	return nil
+}
+
+// Find the set of ledgers tagged as UNDER_DELETE.
+func (provider *storeProvider) getStorageMarkedForDeletion() ([]string, error) {
+	var deleteList []string
+
+	db := provider.dbProvider.GetDBHandle(systemNamespace)
+	defer db.Close()
+
+	val, err := db.Get(underDeletionKey)
+	if err != nil {
+		return nil, errors.WithMessagef(err, "retrieving storage marked for deletion")
+	}
+
+	// no storage previously marked as delete
+	if val == nil {
+		return deleteList, nil
+	}
+
+	if err = json.Unmarshal(val, &deleteList); err != nil {
+		return nil, errors.WithMessagef(err, "unmarshalling json delete list: %s", string(val))
+	}
+
+	return deleteList, nil
+}
+
+// Remove a ledger ID from the list of transient storages currently marked for delete.
+func (provider *storeProvider) clearStorageDeletionStatus(ledgerID string) error {
+	dl, err := provider.getStorageMarkedForDeletion()
+	if err != nil {
+		return errors.WithMessagef(err, "clearing storage flag for ledger [%s]", ledgerID)
+	}
+
+	var updatedList []string
+
+	// retain all entries other than the one to be cleared.
+	for i := 0; i < len(dl); i++ {
+		if !strings.EqualFold(ledgerID, dl[i]) {
+			updatedList = append(updatedList, dl[i])
+		}
+	}
```

Comment:
Here and at another place...
```suggestion
	for _, l := range dl {
		if ledgerID == l {
			continue
		}
		updatedList = append(updatedList, l)
	}
```
- Case-sensitive comparison
- More conventional way of iterating in golang